content/docs/pipelines/reference/api/kubeflow-pipeline-api-spec.html

```diff
@@ -300,29 +319,54 @@ <h3 id="tag-PipelineService" class="swagger-summary-tag">Tag: PipelineService</h
+                        <td><a href="#operation--apis-v1beta1-pipeline_versions-get">GET /apis/v1beta1/pipeline_versions</a></td>
                         <td></td>
                     </tr>
                     <tr>
-                        <td><a href="#operation--apis-v1beta1-pipelines-post">POST /apis/v1beta1/pipelines</a></td>
+                        <td><a href="#operation--apis-v1beta1-pipeline_versions-post">POST /apis/v1beta1/pipeline_versions</a></td>
                         <td></td>
                     </tr>
                     <tr>
-                        <td><a href="#operation--apis-v1beta1-pipelines--id--get">GET /apis/v1beta1/pipelines/{id}</a></td>
+                        <td><a href="#operation--apis-v1beta1-pipeline_versions--version_id--get">GET /apis/v1beta1/pipeline_versions/{version_id}</a></td>
                         <td></td>
                     </tr>
                     <tr>
-                        <td><a href="#operation--apis-v1beta1-pipelines--id--delete">DELETE /apis/v1beta1/pipelines/{id}</a></td>
+                        <td><a href="#operation--apis-v1beta1-pipeline_versions--version_id--delete">DELETE /apis/v1beta1/pipeline_versions/{version_id}</a></td>
                         <td></td>
                     </tr>
                     <tr>
-                        <td><a href="#operation--apis-v1beta1-pipelines--id--templates-get">GET /apis/v1beta1/pipelines/{id}/templates</a></td>
+                        <td><a href="#operation--apis-v1beta1-pipeline_versions--version_id--templates-get">GET /apis/v1beta1/pipeline_versions/{version_id}/templates</a></td>
                         <td></td>
                     </tr>
+                    <tr>
+                        <td><a href="#operation--apis-v1beta1-pipelines-get">GET /apis/v1beta1/pipelines</a></td>
+                        <td><p>Find all pipelines.</p>
+        </td>
+                    </tr>
+                    <tr>
+                        <td><a href="#operation--apis-v1beta1-pipelines-post">POST /apis/v1beta1/pipelines</a></td>
+                        <td><p>Add a pipeline.</p>
+        </td>
+                    </tr>
+                    <tr>
+                        <td><a href="#operation--apis-v1beta1-pipelines--id--get">GET /apis/v1beta1/pipelines/{id}</a></td>
+                        <td><p>Find a specific pipeline by ID.</p>
+        </td>
+                    </tr>
+                    <tr>
+                        <td><a href="#operation--apis-v1beta1-pipelines--id--delete">DELETE /apis/v1beta1/pipelines/{id}</a></td>
+                        <td><p>Delete a pipeline.</p>
+        </td>
+                    </tr>
+                    <tr>
+                        <td><a href="#operation--apis-v1beta1-pipelines--id--templates-get">GET /apis/v1beta1/pipelines/{id}/templates</a></td>
+                        <td><p>Get a YAML template for the selected pipeline.</p>
```

Comment:
Can we give a bit of information about what a template is? Also, does this end point list all templates rather than returning a specific one?

Reply:
Updated as follows:

"Returns a single YAML template that contains the description, parameters, and metadata associated with the pipeline provided."

Tried to describe what is in it since I am not sure what you would actually use the template for. My guess might be portability for custom pipelines between different KF instances (Or other ML platforms?) but I can't verify that just now.